### DISCUSSION TITLE
Suzuka Full Node Bootstrap via Execution

### DISCUSSION THREAD

**Discussion by l-monninger** [movementlabsxyz/movement#471] [Ideas]
# Summary
I'm opening this thread to discuss the possibility of performing Suzuka Full Node bootstrapping by simply executing blocks read from the DA. Historically, this has been deemed intractable for our needs due to the execution time needed. However, platforms such as Avalanche have taken this approach for provisional success.

**Comment by musitdev** [comment]
Yes, we can use the L1 settlement and Celestia to bootstrap a node.
The L1 Settlement contains:
 * the height of the corresponding block,
 * the block_id which is the Celestia block commitment,
 * the commitment, which is the hash of the Aptos state proof after the block execution.
 
 To bootstrap a node from genesis or any trusted height using Celestia, this process can be used:
 * get the L1 commitment at trusted height +1.
 * Use the block_id of the commitment to get the blob from Celestia using the API `Get(height uint64, namespace share.Namespace, commitment blob.Commitment)*blob.Blob`
 * Execute the blob to get an executed block and the resulting state proof.
 * To verify the block execution (Celestia access is not trusted) : use the new state proof to generate its commitment and verify it with the one return from L1.
 * Apply this process until you get the current height.
